examples/data_type2.rs

```diff
@@ -0,0 +1,107 @@
+use raw::KeyType;
```

Comment:
@hahnandrew - thank you for updating the PR.  But could you combine the two data type examples?  I see that Valkey C code has datatype2 but I really don't see why we need to two separate .rs files here as so much code is duplicated.

Reply:
The valkey_module! macro is definable only once a file, and we want to test callbacks for two versions of digest - `ValkeyModule_DigestAddLongLong` and `ValkeyModule_DigestAddStringBuffer`, which requires us to register two versions of `MY_VALKEY_TYPE`. Do note that the two `alloc_set` and `alloc_get` are different in that they return and operate on strings or longlongs

Reply:
Since the recent comments seem to center around this thread let's go ahead and consolidate our discussion here @dmitrypol @KarthikSubbarao.

Reply:
@KarthikSubbarao - I see @hahnandrew points.  Any concerns if I merge the PR?

Reply:
No concerns - we can merge the PR as it is not a blocker for me